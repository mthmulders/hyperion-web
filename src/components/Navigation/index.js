import React from 'react';
import { Link } from 'react-router';

import { Nav, Navbar, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';

const NAV_LINKS = {
    home: { link: '/', title: 'Home' },
    current: { link: '/current', title: 'Current' },
};

class Navigation extends React.Component {
    renderNavItem(linkName) {
        const link = NAV_LINKS[linkName];
        return (
            <LinkContainer to={ { pathname: link.link } } key={ linkName }>
                <NavItem className={ this.props.activePage === linkName ? 'active' : '' } >
                    { link.title }
                </NavItem>
            </LinkContainer>
        );
    }

    render() {
        const brand = <Link to="/" className="navbar-brand">Υπερίων</Link>;
        const links = Object.keys(NAV_LINKS).map((i) => this.renderNavItem(i));

        return (
            <Navbar staticTop componentClass="header" role="banner">
                <Navbar.Header>
                    <Navbar.Brand>{ brand }</Navbar.Brand>
                </Navbar.Header>
                <Nav>
                    { links }
                </Nav>
            </Navbar>
        );
    }
}

Navigation.propTypes = { activePage: React.PropTypes.string.isRequired };

export default Navigation;
