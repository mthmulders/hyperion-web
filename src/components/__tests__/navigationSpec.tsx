import * as React from 'react';

import { shallow } from 'enzyme';

import Navigation from '../navigation';

describe('<Navigation />', () => {
    it('should show menu bar', () => {
        // Act
        const wrapper = shallow(<Navigation />);

        // Assert
        expect(wrapper).toMatchSnapshot();
    });
});
