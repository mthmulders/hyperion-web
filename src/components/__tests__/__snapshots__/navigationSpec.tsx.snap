// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should show menu bar 1`] = `
<nav
  className="navbar navbar-expand-md navbar-dark bg-primary"
>
  <div
    className="container"
  >
    <button
      aria-label="Toggle navigation"
      className="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded="false"
      className="collapse navbar-collapse"
      style={Object {}}
    >
      <ul
        className="md-auto navbar-nav"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/live"
            onClick={[Function]}
          >
            Live
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/recent"
            onClick={[Function]}
          >
            Recent
          </a>
        </li>
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            context={
              Object {
                "direction": "down",
                "disabled": undefined,
                "inNavbar": true,
                "isOpen": false,
                "toggle": [Function],
              }
            }
            href="#"
            onClick={[Function]}
          >
            History
          </a>
          <div
            aria-hidden={true}
            className="navbar-light dropdown-menu"
            context={
              Object {
                "direction": "down",
                "disabled": undefined,
                "inNavbar": true,
                "isOpen": false,
                "toggle": [Function],
              }
            }
            role="menu"
            tabIndex="-1"
          >
            <a
              className="dropdown-item"
              context={
                Object {
                  "direction": "down",
                  "disabled": undefined,
                  "inNavbar": true,
                  "isOpen": false,
                  "toggle": [Function],
                }
              }
              href="/history/daily"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Daily
            </a>
            <a
              className="dropdown-item"
              context={
                Object {
                  "direction": "down",
                  "disabled": undefined,
                  "inNavbar": true,
                  "isOpen": false,
                  "toggle": [Function],
                }
              }
              href="/history/monthly"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Monthly
            </a>
          </div>
        </li>
        <li
          className="dropdown nav-item"
          onKeyDown={[Function]}
        >
          <a
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle nav-link"
            context={
              Object {
                "direction": "down",
                "disabled": undefined,
                "inNavbar": true,
                "isOpen": false,
                "toggle": [Function],
              }
            }
            href="#"
            onClick={[Function]}
          >
            Usage
          </a>
          <div
            aria-hidden={true}
            className="navbar-light dropdown-menu"
            context={
              Object {
                "direction": "down",
                "disabled": undefined,
                "inNavbar": true,
                "isOpen": false,
                "toggle": [Function],
              }
            }
            role="menu"
            tabIndex="-1"
          >
            <a
              className="dropdown-item"
              context={
                Object {
                  "direction": "down",
                  "disabled": undefined,
                  "inNavbar": true,
                  "isOpen": false,
                  "toggle": [Function],
                }
              }
              href="/usage/monthly"
              onClick={[Function]}
              role="menuitem"
              tabIndex="0"
            >
              Monthly
            </a>
          </div>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/about"
            onClick={[Function]}
          >
            About
          </a>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
